src/wikigraph.cc

```diff
@@ -256,12 +346,32 @@ void WikiGraph::displayCentralityProgress(const SafeQueue& queue, const std::vec
   if (progress == 1.0) std::cout << std::endl;
 }
 
-std::vector<std::pair<std::string, double>> WikiGraph::sortCentralityMap(const std::map<std::string, double>& centrality_map) const {
+void WikiGraph::displayPageRankProgress(const size_t& iter, const size_t& total) const {
+  double progress = (double)iter / (double) (total - 1);
+  std::cout << "[ Progess: " ;
+  std::cout << int(progress * 100.0) << "% ] \r";
+  std::cout.flush();
+  if (progress == 1.0) std::cout << std::endl;
+}
+
+std::vector<std::pair<std::string, double>> WikiGraph::getSortedCentrality() {
+  if (centrality_map.empty()) getCentralityMap();
   std::vector<std::pair<std::string, double>> vector(centrality_map.begin(), centrality_map.end());
   std::sort(vector.begin(), vector.end(), 
-  [](const std::pair<std::string, int> &a, const std::pair<std::string, int> &b) {
+  [](const std::pair<std::string, double> &a, const std::pair<std::string, double> &b) {
+    return a.second > b.second;
+  });
+  return vector;
+}
+
+std::vector<std::pair<std::string, double>> WikiGraph::getSortedPageRank() {
+  if (page_rank_map.empty()) rankPages();
+  std::vector<std::pair<std::string, double>> vector(page_rank_map.begin(), page_rank_map.end());
+  std::sort(vector.begin(), vector.end(), 
+  [](const std::pair<std::string, double> &a, const std::pair<std::string, double> &b) {
     return a.second > b.second;
   });
+  std::cout << "sorted" << std::endl;
```

Comment:
is this cout necessary? wondering if it is worth it since a sorted vector means we can go straight to outputting what the user actually wants